packages/engine/src/signature/signature.cairo

```diff
@@ -393,13 +400,33 @@ pub fn parse_pub_key(pk_bytes: @ByteArray) -> Result<Secp256k1Point, felt252> {
 // This function extracts the `r` and `s` values from a DER-encoded ECDSA signature (`sig_bytes`).
 // The function performs various checks to ensure the integrity and validity of the signature.
 pub fn parse_signature(sig_bytes: @ByteArray) -> Result<Signature, felt252> {
-    let mut sig_len: usize = sig_bytes.len() - constants::HASH_TYPE_LEN;
+    if sig_bytes[0] != 0x30 {
+        return Result::Err('invalid sig fmt: no header');
+    }
+    let mut sig_len: usize = sig_bytes[1].into();
+    if sig_len + 2 > sig_bytes.len() || sig_len + 2 < constants::MIN_SIG_LEN {
+        return Result::Err('invalid sig fmt: bad length');
+    }
+
+    let mut start = 0;
+    let sig_bytes = @sub_byte_array(sig_bytes, ref start, sig_len + 2);
+    if sig_bytes[2] != 0x02 {
+        return Result::Err('invalid sig fmt: no 1st marker');
+    }
+
     let mut r_len: usize = sig_bytes[3].into();
+    if r_len <= 0 || r_len > sig_len - 5 {
+        return Result::Err('invalid sig fmt: bogus R length');
+    }
+
     let mut s_len: usize = sig_bytes[r_len + 5].into();
+    if s_len <= 0 || s_len > sig_len - r_len - 4 {
```

Comment:
And would this be sig_len - r_len - 6 since `index` is 6 at this point also : https://github.com/btcsuite/btcd/blob/24eb815168f49dea84767817717a11bd7928eb23/btcec/ecdsa/signature.go#L120?